iOS_SDK/OneSignalSDK/Source/OSMigrationController.m

```diff
@@ -0,0 +1,46 @@
+/**
+Modified MIT License
+
+Copyright 2020 OneSignal
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+1. The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+2. All copies of substantial portions of the Software may only be used in connection
+with services provided by OneSignal.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#import <Foundation/Foundation.h>
+#import "OneSignal.h"
+#import "OSMigrationController.h"
+#import "OSIndirectInfluence.h"
+
+@implementation OSMigrationController
+
+- (void)migrate {
+    [self migrateFromVersion_2_14_0_to_greater];
```

Comment:
What if someone downgrades, shouldn't we revert the name back as well? Or is this too much of a one off case/unexpected behavior?

`[self migrateToVersion_02_13_01_AndLesser];`
`[NSKeyedUnarchiver setClass:[OSIndirectNotification class] forClassName:@"OSIndirectInfluence"];`

Reply:
For this, we will need to save the last version know, in that way we can choose for which migration to use.
@jkasten2 what do you think? should we add version logic and contemplate this case?

Reply:
If someone downgrades to 2.13.1 then there won't be any code there to downgrade. So writing a `migrateToVersion_02_13_01_AndLesser` here for the 2.14.1 release would be impossible to hit its run condision.

There are three options to come to mind to handle downgrading an SDK version:
1. We could make a 2.13.2 release to support downgrading, but we would always need to make sure we make all release with a migration a minor version bump & someone would know this is the only safe way to migrate.
2. Make a migration rollback schema. It will contain things like codeable class renames, field renames, add / removing fields, etc. We will need to support all possible change operations in the first release so when the code is rolled back the generic code can read all the operations it needs to rollback.
3. Clear OneSignal data. This is what we do for the Android SDK today with SQL database, we just clear all records for the database to be safe when the SDK is rolled back.